schema_compatibility.go

```diff
@@ -230,12 +232,192 @@ func (c *SchemaCompatibility) contains(a []string, s string) bool {
 	return false
 }
 
-func (c *SchemaCompatibility) getField(a []*Field, f *Field) (*Field, bool) {
+type getFieldOptions struct {
```

Comment:
It is unclear to me why this is needed. Should it not always resolve the same way?

Reply:
The second call of **`getField`** receives the reader's fields as the first argument instead of the writer's. I agree with you; it would be nice if it resolves the same way.